waiter/src/waiter/service_description.clj

```diff
@@ -1012,14 +1064,19 @@
     "Returns nil if the provided descriptor contains a valid service description.
      Else it returns an instance of Throwable that reflects the validation error."
     [kv-store service-description-builder
-     {:keys [core-service-description passthrough-headers service-description service-id waiter-headers]}]
+     {:keys [core-service-description passthrough-headers service-description service-id sources waiter-headers]}]
     (sling/try+
       (let [stored-service-description (fetch-core kv-store service-id)]
         ; Validating is expensive, so avoid validating if we've validated before, relying on the fact
         ; that we'll only store validated service descriptions
         (when-not (seq stored-service-description)
-          (validate service-description-builder core-service-description {:allow-missing-required-fields? false})
-          (validate service-description-builder service-description {:allow-missing-required-fields? false}))
+          (let [{:keys [profile->overrides]} sources]
+            (validate service-description-builder core-service-description
+                      {:allow-missing-required-fields? false
+                       :profile->overrides profile->overrides})
```

Comment:
Feels like `service-description-builder` should already have this.

Reply:
let's do it in a separate PR, `service-description-defaults` can also move.

Reply:
Unless there's a compelling reason to _not_ do it now, it seems like it makes more sense to inject it in the right place now rather than merge in new code that we already know we want to immediately go back and restructure.

Reply:
Lots of refactoring and adds noise to the PR.
I will do it in a follow-up PR.